src/openms/include/OpenMS/ANALYSIS/TOPDOWN/DeconvolutedSpectrum.h

```diff
@@ -0,0 +1,197 @@
+
+namespace OpenMS
+{
+  class PeakGroup;
+
+  /**
+       @brief A class representing a deconvoluted spectrum.
+       DeconvolutedSpectrum consists of PeakGroups representing masses.
+       For MSn n>1, a PeakGroup representing the precursor mass is also added in this class. Properly assigning a precursor mass
+       from the orgianl precusor peak and its deconvolution result is very important in top down proteomics. This assignment is
+       performed here for conventional datasets. But for FLASHIda acquired datasets, the assignment is already done by FLASHIda.
+       So this class simply use the results from FLASHIda log file for assignment. The parsing of FLASHIda log file is done
+       in FLASHDeconv tool class.
+       It also contains functions to write in different formats and a function to export this class into MSSpectrum.
+  */
+  class OPENMS_DLLAPI DeconvolutedSpectrum :
+      private std::vector<PeakGroup>
+  {
+  public:
+    typedef FLASHDeconvHelperStructs::LogMzPeak LogMzPeak;
+    using std::vector<PeakGroup>::push_back;
+    using std::vector<PeakGroup>::operator[];
+    using std::vector<PeakGroup>::size;
+    using std::vector<PeakGroup>::begin;
+    using std::vector<PeakGroup>::end;
+    using std::vector<PeakGroup>::swap;
+    using std::vector<PeakGroup>::empty;
+    using std::vector<PeakGroup>::reserve;
+    using std::vector<PeakGroup>::clear;
+
+    /// default constructor
+    DeconvolutedSpectrum() = default;
+
+    /**
+       @brief Constructor for DeconvolutedSpectrum. Takes the spectrum and scan number calculated from outside
+       @param spectrum spectrum for which the deconvolution will be performed
+       @param scan_number scan number of the spectrum: this argument is put here for real time case where scan number should be input separately.
+  */
+    explicit DeconvolutedSpectrum(const MSSpectrum &spectrum, const int scan_number);
+
+    /// default deconstructor
+    ~DeconvolutedSpectrum() = default;
+
+    /// copy constructor
+    DeconvolutedSpectrum(const DeconvolutedSpectrum &) = default;
+
+    /// move constructor
+    DeconvolutedSpectrum(DeconvolutedSpectrum &&other) = default;
+
+    /// assignment operator
+    DeconvolutedSpectrum &operator=(const DeconvolutedSpectrum &deconvoluted_spectrum) = default;
+
+    /**
+        @brief write the header in the tsv output file (spectrum level)
+        @param fs file stream to the output file
+        @param ms_level ms level of the spectrum
+        @param detail if set true, detailed information of the mass (e.g., peak list for the mass) is written
+   */
+    static void writeDeconvolutedMassesHeader(std::fstream &fs,
+                                              const int ms_level,
+                                              const bool detail);
+
+    /**
+      @brief write the deconvoluted masses in the output file (spectrum level)
+      @param fs file stream to the output file
+      @param file_name FLASHDeconv paramter
```

Comment:
```suggestion
      @param file_name FLASHDeconv parameter
```
What is this parameter? Not 100% clear from documentation

Reply:
Removed this parameter.